### PR TITLE
Notification draft functionality

### PR DESCRIPTION
**Pull request by we-ai**
This PR implements notification draft functionality, described in [#860](https://github.com/episphere/connect/issues/860). Code cleanups were also done during this process.

### CODE REVIEW

File: siteManagerDashboard/notifications/retrieveNotifications.js [hunk 1]
```diff
@@ -255,9 +263,11 @@ const triggerCategories = (originalCategoriesHolder, categoryName) => {
...
             if (categoryName === 'Filter by Category' || categoryName === tempCategory) {
                 a.innerHTML = e.target.textContent;
-                const siteKey = await getAccessToken();
-                const response = await getNotificationSchema(e.target.textContent, siteKey);
-                mainContent.innerHTML = await render(response);
+                const idToken = await getIdToken();
+                const showDrafts = appState.getState().notification?.showDrafts;
+                const response = await getNotificationSchema(e.target.textContent, idToken, showDrafts);
+                appState.setState({ notification: { showDrafts, schemaArray: response.data } });
+                document.getElementById("mainContent").innerHTML = await render(response.data, showDrafts);
```
Comment (JoeArmani) [inline siteManagerDashboard/notifications/retrieveNotifications.js:270]: is render() async?

File: siteManagerDashboard/notifications/storeNotifications.js [hunk 2]
```diff
@@ -245,68 +252,72 @@ export const mapSchemaNotificaiton = (updateSchemaNotification, concepts, flag)
...
+      schema["conditions"][element.value] = {};
+      if (valueTypeEleArray[i].value === "string") {
+        schema["conditions"][element.value][opeartorEleArray[i].value] = valueEleArray[i].value;
+      } else if (valueTypeEleArray[i].value === "number") {
+        schema["conditions"][element.value][opeartorEleArray[i].value] = parseInt(valueEleArray[i].value);
+      }
+    });
 
-export const addEventMoreCondition = (concepts, flag) => {
+    storeNotificationSchema(schema);
```
Comment (JoeArmani) [inline siteManagerDashboard/notifications/storeNotifications.js:316]: should we await this or does it not matter?

**Review by JoeArmani — APPROVED** [review]
looks good

**Review by ramavavi — APPROVED** [review]

**Comment by ramavavi** [comment]
This branch has merge conflicts.

File: siteManagerDashboard/notifications/storeNotifications.js (hunk 2 above)
Comment (we-ai) [reply siteManagerDashboard/notifications/storeNotifications.js:316]: `await` is needed.

File: siteManagerDashboard/notifications/retrieveNotifications.js (hunk 1 above)
Comment (we-ai) [reply siteManagerDashboard/notifications/retrieveNotifications.js:270]: It's not.

**Pull request merged**